run_process_zephir_incremental.sh

```diff
@@ -22,6 +22,23 @@ cd $TMPDIR
 SCRIPTNAME=`basename $0`
 zephir_date="$(echo $YESTERDAY | sed 's/\(....\)\(..\)/\1-\2-/')"
 
+# Route all external processes through these functions
+# to avoid silent failures.
+function run_external_command {
+  $1
```

Comment:
Using `$1` can lead to quoting problems. I would recommend instead using `"$@"` (with the double quotes) which will preserve all the original parameters as quoted.

and then later down we would do (without needing additional quoting)

```bash
run_external_command $ROOTDIR/ftpslib/ftps_zephir_get exports/$ZEPHIR_VUFIND_EXPORT $ZEPHIR_VUFIND_EXPORT
```

Reply:
It's looking like we need `eval "$@"`(or maybe just `eval $@`?) just for the cases where we try to run something with an environment:
```
run_external_command SOME_ENV=blah ls | wc
```
Is this acceptable for the general case?

Reply:
Looking at the documentation for `eval` I agree that is the best option here. I'm unsure of the shell parsing for the particular example you gave (I think the `| wc` part would not be passed to `run_external_command`), but I think that's OK. We don't need to support arbitrary complexity, just act in whatever fashion is least surprising, and I think `eval "$@"` will get us there.

Reply:
I've tried to quote any commands that use pipes so the whole bucket chain should be checked. Possible rabbit hole into `pipefail` not fallen into, yet anyway. Crossing my fingers this will get us where we need to be.